### ISSUE TITLE
data_protection_backup_policy_storageblobs

### ISSUE THREAD

**Issue by stravze** [terraform-providers/terraform-provider-azurerm#12318]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

With the recent release of the new service data_protection

We need backup policy for Azure Blob Storage

https://github.com/terraform-providers/terraform-provider-azurerm/pull/12220

### New or Affected Resource(s)

* azurerm_data_protection_backup_policy

### Potential Terraform Configuration

```hcl
resource "azurerm_data_protection_backup_policy_storageblob" "example" {
  name                = "example-backup-policy"
  resource_group_name = azurerm_resource_group.rg.name
  vault_name          = azurerm_data_protection_backup_vault.example.name

  backup_rule {
    repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]               // when to do the backup 
  }

  default_retention_duration = "P4M"             // default retention duration, which is applied when no other retention rule

  retention_rule {
    name             = "weekly"
    duration         = "P6M"                       // retention duration
    tagging_priority = 20                           // if a backup matches more than 1 retention rule's criteria, it will be applied with the retention rule which has higher priority 
    tagging_criteria {                                 // which backup will be applied this retention rule
      absolute_criteria = "FirstOfWeek"   // first successful backup of the week/month/year
    }
  }
```

### References

* #0000

**Comment by ms-henglu** [comment]
I'm working on it. Policy and Instance for both storage blob and disk.

**Comment by tombuildsstuff** [comment]
Fixed via #12362
